tests/testData/annotation/call_input_checking.wdl

```diff
@@ -0,0 +1,238 @@
+version 1.0
+
+import "nonexistent_import.wdl" as nonexistent_import
+
+workflow my_workflow {
+
+    # === NO INPUTS ===
+
+    # No inputs, none provided (correct)
+    call no_inputs
+
+    # No inputs, empty scope for input block (highlight)
+    <error descr="Task does not take inputs.">call no_inputs {
+
+    }</error>
+
+    # No inputs, empty input list (highlight)
+    # IMO: this one is a bit debatable as to whether it should highlight, since the list IS technically empty (AEN, 2018-07-17)
+    <error descr="Task does not take inputs.">call no_inputs {
+        input:
+    }</error>
+
+    # No inputs, full input block provided (highlight)
+    <error descr="Task does not take inputs.">call no_inputs {
+        input:
+            a = "a",
+            b = 5,
+            c = false
+    }</error>
+
+
+    # === REQUIRED INPUTS ===
+
+    # Required inputs provided (correct)
+    call required_inputs {
+         input:
+            a = "a",
+            b = 5,
+            c = false
+    }
+
+    # Required inputs provided, reordered (correct)
+    call required_inputs {
+        input:
+            b = 5,
+            a = "a",
+            c = false
+    }
+
+    # All required inputs provided, plus extraneous (highlight)
+    <error descr="Unexpected inputs(s) for task: d">call required_inputs {
+        input:
+            a = "a",
+            b = 5,
+            c = false,
+            d = "Zardoz"
+    }</error>
+
+    # Wrong input name (highlight; two annotations)
+    <error descr="Missing required inputs(s) for task: a"><error descr="Unexpected inputs(s) for task: aa">call required_inputs {
+        input:
+            aa = "a",
+            b = 5,
+            c = false
+    }</error></error>
+
+    # No input block (highlight)
+    <error descr="Task has required inputs.">call required_inputs</error>
+
+    # Empty input scope (highlight)
+    <error descr="Missing required inputs(s) for task: a, b, c">call required_inputs {
+
+    }</error>
+
+    # Empty input list (highlight)
+    <error descr="Missing required inputs(s) for task: a, b, c">call required_inputs {
+        input:
+    }</error>
+
+    # Missing inputs (highlight)
+    <error descr="Missing required inputs(s) for task: b, c">call required_inputs {
```

Comment:
A "missing input" is not an error and should be highlighted as a warning, with text something like:
> Unstable workflow interface: inputs 'File a' and 'String b' to call 'foo' are not provided. To use this workflow as a subworkflow, you'll need to pass through inputs for these values.

Side note: this message would be the place to add the auto-completion option in #94.

Reply:
Is it wise to show this warning if we don't give the user any way to make it go away?

We can't exactly check the user's input JSON.